### DISCUSSION TITLE
Can I flash miryoku QMK onto my keychron which is not on the list of supported keyboard in QMK?

### DISCUSSION THREAD

**Discussion by huydinhle** [manna-harbour/miryoku#196] [Q&A] [answered]
pretty much the title? and if yes I can, then how would I able to do that?

**Comment by manna-harbour** [comment, marked as answer]
For a keyboard to be supported by Miryoku QMK it requires the following:

1. The keyboard is supported in QMK.
2. The keyboard has a compatible physical layout.
3. A mapping file exists in Miryoku QMK to map the Miryoku keymap layout onto the keyboard's keymap layout.

The following Keychron keyboards are supported in QMK: https://github.com/qmk/qmk_firmware/tree/master/keyboards/keychron. Which keyboard do you have?

They seem to generally be row-stagger keyboards 60% and larger. Miryoku is designed for ergo keyboards. There are a few approaches to using Miryoku on row-stagger. See https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#60_ansi. Out of those 3, only the `lite` mapping is recommended for general use. Keyboards with more thumb keys could also support a complete mapping without needing to shift rows and columns.

The only mapping for row stagger currently in Miryoku QMK is https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#60_ansi. If your keyboard supports that community layout then you can build Miryoku QMK. No Keychron keyboards currently support that community layout. `q60` supports a similar layout and if the layout is identical only the keyboard's `info.json` would need to be modified to enable the community layout. Any other keyboard would need a corresponding mapping file to be created, equivalent to https://github.com/manna-harbour/miryoku_qmk/blob/miryoku/layouts/community/60_ansi/manna-harbour_miryoku/config.h but modified for the target layout.

Alternatively, Miryoku KMonad can be used on row stagger in a similar fashion. See https://github.com/manna-harbour/miryoku_kmonad. It doesn't depend on the exact physical layout, only the keys in use in the mapping, and if necessary the mapping can be adjusted by editing https://github.com/manna-harbour/miryoku_kmonad/blob/main/src/miryoku_kmonad.kbd.cpp.

**Comment by opestov** [comment]
[Here is](https://opestov.github.io/k3/) concrete steps for Keychron K3 Pro.
